File: Python/NPTEL/Amstrong.py
a=input("Enter a number")
a=int(a)
org=a
p=0
sum=0
while(a!=0):
    rem=a%10
    p=p+1
    a=a//10 #Here // operator performs integer devision
a=org
while(a!=0):
    rem=a%10
    sum+=rem**p
    a=a//10 #Here // operator performs integer devision
if(sum==org):
    print(org,"is an amstrong number")
else:
    print(org,"is not an amstrong number")

File: C Programming/PIC Tutorial/Amstrong.c
//Amstrong check for no.s between 100 & 1000
#include<stdio.h>
#include<math.h>
void main()
{
    int a,rem,org,p=0,sum=0;
    printf("Enter a number");
    scanf("%d",&a);
    org=a;
    while(a!=0)
    {
        rem=a%10;
        p++;
        a=a/10;
    }
    a=org;
    while(a!=0)
    {
        rem=a%10;
        sum=sum+pow(rem,p);
        a=a/10;
    }
    
    if(org<100||org>1000)
    {printf("invalid");}
    else if(sum==org)
    {printf("Amstrong no.");}
    else
    {printf("Not amstrong no.");}
    
}